src/map.cpp

```diff
@@ -3754,6 +3754,50 @@ bool map::hit_with_acid( const tripoint &p )
     return true;
 }
 
+void map::fell_tree( const tripoint &p, const tripoint &direction )
+{
+    int dir_x, dir_y;
+
+    if( direction == tripoint_zero ) {
+        // TODO: pick a random direction properly
+        dir_x = ( 2 * rng( 0, 1 ) - 1 );
+        dir_y =  rng( -1, 1 );
+    } else {
+        // an extra-long direction vector shouldn't mean a bigger tree
+        dir_x = ( direction.x > 0 ) - ( direction.x < 0 );
+        dir_y = ( direction.y > 0 ) - ( direction.y < 0 );
```

Comment:
That's an awkward way of normalizing magnitudes, which can also be rather inaccurate. As for now, the function accepts already normalized points, so there's no immediate need for it, but if you want it to look nice and tidy, you could use angles instead:

```c++
void map::fell_tree( const tripoint &p, int angle )
{
    tripoint to;
    // Angle normalization may be required at this point.
    calc_ray_end( angle, 3 + rng( -1, 1 ), p, to );
    for( const auto &elem : line_to( p, to, rng( 1, 8 ) ) ) {
        ...
    }
}

fell_tree( p, rng( 0, 360 ) );
fell_tree( p, coord_to_angle( {}, dir ) + rng( -30, 30 ) );

```

Reply:
That's pretty slick!

It looks like `calc_ray_end` depends on the `trigdist` setting, though, which would mean smaller yields for diagonal directions if `trigdist` is set. (1-2 trunks instead of 2-4).

Reply:
> It looks like calc_ray_end depends on the trigdist setting, though

It does indeed. As a solution, it could probably be split for better reusability (like `trig_dist`, `square_dist`, and `rl_dist` are).